packages/backend/src/payment-method/ilp/connector/core/middleware/balance.ts

```diff
@@ -40,11 +40,18 @@ export function createBalanceMiddleware(): ILPMiddleware {
     }
 
     const sourceAmount = BigInt(amount)
-    const destinationAmountOrError = await services.rates.convertSource({
-      sourceAmount,
-      sourceAsset: accounts.incoming.asset,
-      destinationAsset: accounts.outgoing.asset
-    })
+    const token = accounts.outgoing.http?.outgoing.authToken
+    const peer = token
+      ? await services.peers.getByIncomingToken(token)
+      : undefined
```

Comment:
Each of the `accounts` here are either a Peer, Incoming Payment, or Outgoing Payment. Each of them should already have a tenantId on them, so if we add `tenantId` as a property of the `ConnectorAccount` interface, you should be able to use it directly.